atkinson/errors/reporters/generic.py

```diff
@@ -0,0 +1,65 @@
+#! /usr/bin/env python
+""" Python logging based reporter """
+
+from atkinson.errors.reporters import BaseReport
+from atkinson.logging.logger import getLogger
+
+
+class GenericReporter(BaseReport):
+    """ Class for a generic reporter based on the logging module """
+
+    def __init__(self, title, logger):
+        """
+        Class constructor
+        :param logger: The logger to use
+        """
+        self.__title = title
+        self.__log = logger
+
+    @classmethod
+    def new(cls, title, description, config):
+        """
+        Create a new report
+        :param title: The title for the report
+        :param description: A description for the report
+        :param config: Configuration dictionary for the report
+        :return: A LoggingReporter instance
+        """
+        log = getLogger()
+        log.error(f"{title}: {description}")
+        return cls(title, log)
+
+    @classmethod
+    def get(cls, report_id, config):
+        """
+        Retrieve an active report
+        :param report_id: The unique id (the report title) for the report
+        :param config: Configuration dictionary for the report
+        :return: A LoggingReporter instance
+        """
+        return cls(report_id, getLogger())
+
+    @property
+    def report_id(self):
+        """
+        Return the unique id for this report
+        """
+        return self.__title
+
+    def update(self, **kwargs):
+        """
+        Update the report
+        :param kwargs: A dictionary of named arguments to report
+        """
+        if self.__log:
+            message = f"{self.__title}:"
+            for key, value in kwargs.items():
+                message += f"\n\t{key}: {value}"
```

Comment:
Here is the example you requested:
```
message += [f"\n\t{key}: {value}" for key, value in kwargs.items()]
```

If you were trying to do it all in one line for assigning message, you could probably do something like:
```
message = f"{self.__title}: " + "\n\t".join([f"{key}: {value}" for key, value in kwargs.items()])
```

Reply:
I went with a variation of option 2. Unfortunately option one throws a TypeError, since you can contact a list and a string.

Reply:
Yeah, looks like I forgot the join in the first example, sorry - I like the one you went with though!